backend/TreeOfAKind.Domain/Trees/Rules/ThereAreOnlyOneSpouseRule.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections.Generic;
+using System.Linq;
+using TreeOfAKind.Domain.SeedWork;
+using TreeOfAKind.Domain.Trees.People;
+
+namespace TreeOfAKind.Domain.Trees.Rules
+{
+    public class ThereAreOnlyOneSpouseRule : IBusinessRule
+    {
+        public IReadOnlyCollection<Relation> Relations { get; }
+        public PersonId Person1 { get; }
+        public PersonId Person2 { get; }
+        public RelationType RelationType { get; }
+        public ThereAreOnlyOneSpouseRule(IReadOnlyCollection<Relation> relations, PersonId person1, PersonId person2, RelationType relationType)
+        {
+            Relations = relations;
+            Person1 = person1;
+            Person2 = person2;
+            RelationType = relationType;
+        }
+
+        public bool IsBroken()
+        {
+            if (RelationType != RelationType.Spouse) return false;
+            var spouseRelations = Relations.Where(r => r.RelationType == RelationType.Spouse);
+            var spouseOfFirstPerson = spouseRelations.FirstOrDefault(r => r.From == Person1);
+            var spouseOfSecondPerson = spouseRelations.FirstOrDefault(r => r.From == Person2);
```

Comment:
Is every relation represented twice in `Relations`?

Reply:
For u it doesn't matter, what matters is:

Tree people A, B, and C

Add relation spouse AB
Add relation spouse CB
B has two spouses

But yes I do use symmetric spouse relations for simplicity 🙄